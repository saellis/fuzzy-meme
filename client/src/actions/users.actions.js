import * as _ from '../constants/users.actions.constants';

export const createUserAction = (un, pw) => {
	return (dispatch)=>{
		dispatch({type:_.CREATE_USER_PENDING});
		return fetch('http://localhost:3001/users/create', {
			method: 'post',
			headers: {
				'Content-type': 'application/x-www-form-urlencoded; charset=UTF-8'
			},
			body: `username=${un}&password=${pw}`
		}).then(res => res.json()).then(
				data => {
					if(data.err){
						dispatch({type:_.CREATE_USER_ERROR, error: data.err});
					}else{
						dispatch({type:_.CREATE_USER_SUCCESS, data: data});
						dispatch({type:_.CLEAR_CREATE_USER_SYNTAX_ERROR});
					}

				},
				error => dispatch({type:_.CREATE_USER_ERROR})
				//maybe dispatch an action that does some sort of notification on screen?
			);
	};
};

export const setCreateUserSyntaxError = (errors) => {
	return (dispatch) => {
		dispatch({type:_.UPDATE_CREATE_USER_SYNTAX_ERROR, errors:errors});
	};
};

export const clearCreateUserSyntaxError = () => {
	return (dispatch) => {
		dispatch({type:_.CLEAR_CREATE_USER_SYNTAX_ERROR});
	};
};

export const clearCreateUser = () => {
	return dispatch => dispatch({type:_.CLEAR_CREATE_USER})
}

export const loginAction = (un, pw) => {
	return (dispatch)=>{
		dispatch({type:_.LOGIN_PENDING});
		return fetch('http://localhost:3001/users/auth', {
			method: 'post',
			headers: {
				'Content-type': 'application/x-www-form-urlencoded; charset=UTF-8'
			},
			body: `username=${un}&password=${pw}`
		}).then(res => res.json()).then(
				data => {
					//TODO: better check here for success, maybe pass {success: true} from back end
					if(data._id){
						dispatch({type:_.LOGIN_SUCCESS, data: data});
						dispatch({type:_.CLEAR_LOGIN_ERROR_TEXT});
					}else{
						dispatch({type:_.LOGIN_INCORRECT, data: data});
					}
				},
				error => dispatch({type:_.LOGIN_ERROR})
				//maybe dispatch an action that does some sort of notification on screen?
			);
	};
};

export const clearLogin = () => {
	return dispatch => dispatch({type:_.CLEAR_LOGIN});
}
