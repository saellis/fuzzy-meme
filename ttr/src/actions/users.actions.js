export const CREATE_USER_PENDING = 'CREATE_USER_PENDING';
export const CREATE_USER_SUCCESS = 'CREATE_USER_SUCCESS';
export const CREATE_USER_ERROR = 'CREATE_USER_ERROR';



export const createUserAction = () => {
	return (dispatch)=>{
		dispatch({type:CREATE_USER_PENDING});
		return fetch('http://localhost:3001/users/create', { 
		        method: 'POST',
		        data: {
		          username: 'jared',
		          password: 'swinney'
		        }
		      }).then(res => res.json()).then(
				data => dispatch({type:CREATE_USER_SUCCESS, data: data}),
				error => dispatch({type:CREATE_USER_ERROR})
				//maybe dispatch an action that does some sort of notification on screen?
			);
	}
}


export const LOGIN_PENDING = 'LOGIN_PENDING';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_INCORRECT = 'LOGIN_INCORRECT';
export const LOGIN_ERROR = 'LOGIN_ERROR';



export const loginAction = (un, pw) => {
	return (dispatch)=>{
		dispatch({type:LOGIN_PENDING, data: {un:un, pw:pw}});
		return fetch('http://localhost:3001/users/login', { 
		        method: 'POST',
		        data: {
		          username: un,
		          password: pw
		        }
		      }).then(res => res.json()).then(
				data => {
					//TODO: Check to see if successful login
					if(true){
						dispatch({type:LOGIN_SUCCESS, data: data})
					}else{
						dispatch({type:LOGIN_INCORRECT, data: data})
					}
				},
				error => dispatch({type:LOGIN_ERROR})
				//maybe dispatch an action that does some sort of notification on screen?
			);
	}
}

export const UPDATE_FIELD = 'UPDATE_FIELD';

export const updateFieldAction = (key, text) => {
	return (dispatch)=>{
		dispatch({
			type:UPDATE_FIELD,
			text: text,
			key: key
		})
	}
}

