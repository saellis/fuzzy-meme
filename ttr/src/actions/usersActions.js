export const CREATE_USER_PENDING = 'CREATE_USER_PENDING';
export const CREATE_USER_SUCCESS = 'CREATE_USER_SUCCESS';
export const CREATE_USER_ERROR = 'CREATE_USER_ERROR';



export const createUserAction = () => {
	return (dispatch)=>{
		dispatch({type:CREATE_USER_PENDING});
		return fetch('http://localhost:3001/users/create', { 
		        method: 'POST',
		        data: {
		          username: 'jared',
		          password: 'swinney'
		        }
		      }).then(res => res.json()).then(
				data => dispatch({type:CREATE_USER_SUCCESS, data: data}),
				error => dispatch({type:CREATE_USER_ERROR})
			);
	}
}