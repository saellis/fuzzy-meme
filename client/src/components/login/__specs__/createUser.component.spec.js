import { CreateUser } from '../createUser.component.jsx';
import { LoginFieldContainer } from '../../../containers/login/loginField.container.jsx';

import React from 'react';
import { mount, shallow } from 'enzyme';
import { createUserAction }  from '../../../actions/login/createUser.actions';
import { Button } from 'react-bootstrap';

import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe.only('<CreateUser>', () => {
	var wrapper;
	var props;

	beforeEach(() => {
		props = {
			createUser: jest.fn(),
			setErrorText: jest.fn(),
			clearErrorText: jest.fn(),
			errors: ['error'],
			fields: {}
		};
		wrapper = renderer.create(<Provider store={mockStore({})}><CreateUser {...props}/></Provider>);
	});

	test('~~snapshot', () => {
		expect(wrapper.toJSON()).toMatchSnapshot();
	});
	describe('functionality', () => {
		beforeEach(() => {
			wrapper = shallow(<CreateUser {...props}/>);
		});

		test('should save on keypress', () => {
			wrapper.find(LoginFieldContainer).forEach((field) => {
				field.props().textChange(field.props().id, 'test');
				expect(wrapper.state().fields[field.props().id]).toBe('test');
			});
		});

		describe('form validation', () => {
			var field;
			beforeEach(() => {
				wrapper.setState((prevState) => {
					let newState = prevState;
					newState.fields['createUsername'] = 'abcabcabc';
					newState.fields['createPassword'] = 'Aa111111';
					newState.fields['createConfirmPassword'] = 'Aa111111';
					return newState;
				});
				field = wrapper.find(Button);
			});

			test('correct data', () => {
				field.at(0).simulate('click');
				expect(props.createUser.mock.calls.length).toBeGreaterThan(0);
			});

			test('invalid username', () => {
				wrapper.setState((prevState) => {
					let newState = prevState;
					newState.fields['createUsername'] = 'A1';
					return newState;
				});
				field.at(0).simulate('click');
				expect(props.setErrorText.mock.calls.length).toBeGreaterThan(0);
			});

			test('non-matching passwords', () => {
				wrapper.setState((prevState) => {
					let newState = prevState;
					newState.fields['createPassword'] = 'AA111111';
					newState.fields['createConfirmPassword'] = 'AA112111';
					return newState;
				});
				field.at(0).simulate('click');
				expect(props.setErrorText.mock.calls.length).toBeGreaterThan(0);
			});

			test('no lower case', () => {
				wrapper.setState((prevState) => {
						let newState = prevState;
						newState.fields['createPassword'] = 'AA111111';
						newState.fields['createConfirmPassword'] = 'AA111111';
						return newState;
					});
				field.at(0).simulate('click');
				expect(props.setErrorText.mock.calls.length).toBeGreaterThan(0);
			});

			test('password too short', () => {
				wrapper.setState((prevState) => {
						let newState = prevState;
						newState.fields['createPassword'] = 'AA1111';
						newState.fields['createConfirmPassword'] = 'AA1111';
						return newState;
					});
				field.at(0).simulate('click');
				expect(props.setErrorText.mock.calls.length).toBeGreaterThan(0);
			});

			test('no uppercase', () => {
				wrapper.setState((prevState) => {
						let newState = prevState;
						newState.fields['createPassword'] = 'aa111111';
						newState.fields['createConfirmPassword'] = 'aa111111';
						return newState;
					});
				field.at(0).simulate('click');
				expect(props.setErrorText.mock.calls.length).toBeGreaterThan(0);
			});

			test('no digit', () => {
				wrapper.setState((prevState) => {
						let newState = prevState;
						newState.fields['createPassword'] = 'aAaAaAaA';
						newState.fields['createConfirmPassword'] = 'aAaAaAaA';
						return newState;
					});
				field.at(0).simulate('click');
				expect(props.setErrorText.mock.calls.length).toBeGreaterThan(0);
			});
		});
	})

});
