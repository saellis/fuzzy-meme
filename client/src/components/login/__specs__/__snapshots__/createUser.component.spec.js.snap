// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateUser> ~~snapshot 1`] = `
<div>
  <div
    className="col-lg-6 col-lg-offset-3 col-md-6 col-md-offset-3 col-sm-10 col-sm-offset-1 col-xs-10 col-xs-offset-1"
  >
    <div
      className="panel panel-primary"
      id={undefined}
    >
      <div
        className="panel-body"
      >
        <div
          className="form-group"
        >
          <div
            className="margin-bottom-5 col-lg-4 col-md-4 col-sm-2 col-xs-2"
          >
            <label
              className="control-label"
              htmlFor={undefined}
            >
              Username: 
            </label>
          </div>
          <div
            className="margin-bottom-5 col-lg-8 col-md-8 col-sm-10 col-xs-10"
          >
            <input
              className="form-control"
              id={undefined}
              onChange={[Function]}
              placeholder="Username"
              type="text"
            />
          </div>
        </div>
        <div
          className="form-group"
        >
          <div
            className="margin-bottom-5 col-lg-4 col-md-4 col-sm-2 col-xs-2"
          >
            <label
              className="control-label"
              htmlFor={undefined}
            >
              Password: 
            </label>
          </div>
          <div
            className="margin-bottom-5 col-lg-8 col-md-8 col-sm-10 col-xs-10"
          >
            <input
              className="form-control"
              id={undefined}
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
          </div>
        </div>
        <div
          className="form-group"
        >
          <div
            className="margin-bottom-5 col-lg-4 col-md-4 col-sm-2 col-xs-2"
          >
            <label
              className="control-label"
              htmlFor={undefined}
            >
              Confirm Password: 
            </label>
          </div>
          <div
            className="margin-bottom-5 col-lg-8 col-md-8 col-sm-10 col-xs-10"
          >
            <input
              className="form-control"
              id={undefined}
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
            />
          </div>
        </div>
        <button
          className="btn-primary btn btn-default btn-block"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Create 
        </button>
      </div>
    </div>
  </div>
  <div
    className="col-lg-3 col-lg-offset-0 col-md-3 col-md-offset-0 col-sm-10 col-sm-offset-1 col-xs-10 col-xs-offset-1"
  >
    <div
      className="panel panel-danger"
      id={undefined}
    >
      <div
        className="panel-heading"
      >
        Please fix:
      </div>
      <div
        className="panel-body"
      >
        <ul
          className="errors"
        >
          <li
            className="errors"
          >
            error
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
